skimage/transform/_thin_plate_splines.py

```diff
@@ -0,0 +1,316 @@
+        """Estimate the transformation from a set of corresponding points.
+
+        Parameters
+        ----------
+        coords : (N, 2) array_like
+            x, y coordinates to transform
+
+        Returns
+        -------
+        transformed_coords: (N, D) array
+            Destination coordinates
+        """
+        if self.spline_mappings is None:
+            raise ValueError(f"{self.spline_mappings}. Compute the `estimate`")
+        coeffs = self.spline_mappings
+        coords = np.array(coords)
+
+        if not coords.ndim == 2 or coords.shape[1] != 2:
+            raise ValueError("Input `coords` must have shape (N, 2)")
+
+        radial_dist = self._radial_distance(coords[:, 0], coords[:, 1])
+
+        x_warp = self._spline_function(
+            coords[:, 0], coords[:, 1], radial_dist, coeffs[:, 0]
+        )
+        y_warp = self._spline_function(
+            coords[:, 0], coords[:, 1], radial_dist, coeffs[:, 1]
+        )
+        return np.vstack([x_warp, y_warp]).T
+
+    @property
+    def inverse(self):
+        raise NotImplementedError("This is yet to be implemented.")
+
+    def estimate(self, src, dst):
+        """Estimate optimal spline mappings that describe the deformation of the points.
+
+
+        Parameters
+        ----------
+        src : (N, 2) array_like
+            Control point at source coordinates
+        dst : (N, 2) array_like
+            Control point at destination coordinates
+
+        Returns
+        -------
+        success: bool
+            True indicates that the estimation was successful.
+
+        Notes
+        -----
+        -  The number of source and destination points must match (N).
+        """
+
+        check_nD(src, 2)
+        check_nD(dst, 2)
+
+        if len(src) < 3 or len(dst) < 3:
+            raise ValueError(f"{src} points less than 3 is considered undefined.")
+
+        if src.shape != dst.shape:
+            raise ValueError(
+                "The shape of source and destination coordinates must match."
+            )
+
+        self.src = src
+        n, d = src.shape
+
+        dist = sp.spatial.distance.cdist(self.src, self.src)
+        K = _radial_basis_kernel(dist)
+        P = np.hstack([np.ones((n, 1)), src])
+        L = np.zeros((n + 3, n + 3), dtype=np.float32)
+        L[:n, :n] = K
+        L[:n, -3:] = P
+        L[-3:, :n] = P.T
+        V = np.concatenate([dst, np.zeros((d + 1, d))])
+        self.spline_mappings = np.dot(np.linalg.inv(L), V)
+        return True
+
+    def _radial_distance(self, x, y):
+        """Compute the radial distance."""
+        Pi_x = self.src[:, 0]
+        Pi_y = self.src[:, 1]
+
+        dx = Pi_x[:, np.newaxis] - x
+        dy = Pi_y[:, np.newaxis] - y
+        r = np.sqrt(dx**2 + dy**2)
+        radial_dist = _radial_basis_kernel(r)
+        return radial_dist
+
+    def _spline_function(self, x, y, radial_dist, coeffs):
+        """Solve the spline function in the X and Y directions"""
```

Comment:
The spline is a function, but you solve an *equation*... Maybe for this function to be zero? To be minimal?

Reply:
Would "Calculate the value of a spline function in the X and Y directions" be more appropriate.
or "Compute the value of the thin-plate spline function"
?

Reply:
Maybe "Estimate / fit the spline function in Y and Y directions."?